### ISSUE TITLE
SMTP authentication method(loginauth)

### ISSUE THREAD

**Issue by cfclong** [golang/go#40817]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13.4 windows/amd64
</pre>

### Does this issue reproduce with the latest release?
yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\15737\AppData\Local\go-build
set GOENV=C:\Users\15737\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=C:\Users\15737\go
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=D:\Go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=D:\Go\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=C:\Users\15737\AppData\Local\Temp\go-build883883038=/tmp/go-build -gno-record-gcc-switches
</pre></details>

### What did you do?

I just added an authentication method: loginAuth

### What did you expect to see?
I hope that the official can provide the authentication method of SMTP

### What did you see instead?
Lack of an authentication method of SMTP

**Comment by davecheney** [comment]
Would you be able to link to the rfc that describes loginAuth authentication. Thank you

**Comment by dmitshur** [comment]
Is this issue about the `net/smtp` package or something else?

**Comment by jandersonbt** [comment]
I am requesting that this issue be reopened and implemented.  The AUTH LOGIN standard is very much alive as was pointed out by @david7482. AUTH PLAIN is actually [no longer supported](https://support.microsoft.com/en-us/office/outlook-com-no-longer-supports-auth-plain-authentication-07f7d5e9-1697-465f-84d2-4513d4ff0145?ui=en-us&rs=en-us&ad=us) by outlook.com, as I just discovered with a project I am currently working on. Office365 SMTP servers don't support it either.  I was able to easily add AUTH LOGIN to my project following this [post](https://stackoverflow.com/questions/42305763/connecting-to-exchange-with-golang), but I completely agree with previous comments that this should be supported in the `net/smtp` package.  Below is a great reference for how the different types of SMTP authentication work. Between this and the other links I have provided someone should be able to easily implement a AUTH LOGIN method into the `net/smtp` package.
https://www.samlogic.net/articles/smtp-commands-reference-auth.htm
